packages/table/src/Table.tsx

```diff
@@ -35,19 +76,77 @@ export default function Table<Shape>({
   className,
   ...rest
 }: TableProps<Shape>) {
+  const [scrollState, setScrollState] = React.useState<ScrollState>(
+    ScrollState.None,
+  );
+  const divRef = React.useRef<HTMLDivElement>(null);
+  const viewportSize = useViewportSize();
+  console.log(viewportSize);
+
+  useIsomorphicLayoutEffect(() => {
+    const ref = divRef.current;
+
+    if (ref != null && ref.scrollWidth > ref.clientWidth) {
+      setScrollState(ScrollState.Right);
+    } else if (
+      ref != null &&
+      viewportSize != null &&
+      ref.getBoundingClientRect().width < viewportSize.width
+    ) {
+      setScrollState(ScrollState.None);
+    }
+  }, [viewportSize]);
+
+  const handleScroll = (e: React.UIEvent) => {
```

Comment:
I don't think we need the type assertions below if this is changed to`React.UIEvent<HTMLElement>`

Reply:
Even with this React.UIEvent<HTMLElement> I get errors when I remove the assertions below

Reply:
It seems to be because the `T` in `React.UIEvent<T>` is the element the handler is attached to, but for events in general the actual target could be a child of that element. Since this is a scroll event, the target probably can't ever be a child, but the typings don't reflect that. One way to work around it is to type the event as `React.UIEvent & {target: HTMLElement}`, though ideally the React typings would itself define something like `type ScrollUIEvent<T extends Element> = React.UIEvent<T> & {target: T};`

Reply:
```Type 'UIEvent<HTMLDivElement, UIEvent>' is not assignable to type '{ target: HTMLDivElement; }'.
        Types of property 'target' are incompatible.```